Doc/library/typing.rst

```diff
@@ -141,6 +144,17 @@ Type aliases are useful for simplifying complex type signatures. For example::
 Note that ``None`` as a type hint is a special case and is replaced by
 ``type(None)``.
 
+The :keyword:`type` statement is new in Python 3.12. For backwards
+compatibility, type aliases can also be created through simple assignment::
+
+   Vector = list[float]
+
+Or marked with :data:`TypeAlias`::
+
+   from typing import TypeAlias
+
+   Vector: TypeAlias = list[float]
```

Comment:
I know what you mean, but `TypeAlias` is still going to be with us for a while yet. A lot of libraries won't be able to use the new syntax until 3.11 goes EOL.

Reply:
The new wording's better, thanks, and I think you're right that this isn't the place for an extended example here. This bit looks good now.

I _do_, however, think we should add a better example to the section on `typing.TypeAlias` for why you would want to use it (that could be a separate PR, perhaps):
- There's basically nothing there at the moment
- For a lot of users, it will be a long time before they're able to use `typing.TypeAliasType`
- It's a useful feature that a lot of typing users still don't properly understand
- The introduction of a new feature in 3.12, that works in quite a different way (but has almost exactly the same name), only increases the need for this feature to be well documented, or users will get hopelessly confused between the two, I think.

> The docs for the match statement don't come with lots of guidance about how you should be using if statements if you're still on 3.9.

Sure, but that's not an accurate analogy -- a more accurate analogy would be if `if`-statements weren't called if-statements, they were called "matching patterns" or something confusingly similar to pattern-matching :)

> Similarly, users who still support 3.11 or lower should look at the 3.11 docs to see what's possible.

I agree it would be nice if users did that, but my instinct is most people just go to visit https://docs.python.org/3/library/typing.html when they want to find out about `typing`, rather than https://docs.python.org/3.10/library/typing.html to find out about the stuff that's specific to Python 3.10.

Reply:
Let's extend https://docs.python.org/3.12/library/typing.html#typing.TypeAlias in a separate PR then.

I agree the naming does make all this more confusing!